service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/delete/DeleteAzureDiskStep.java

```diff
@@ -0,0 +1,84 @@
+package bio.terra.workspace.service.resource.controlled.flight.delete;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.flight.create.CreateAzureDiskStep;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.resourcemanager.compute.ComputeManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.UUID;
+
+/**
+ * A step for deleting a controlled Azure Disk resource. This step uses the following process to
+ * actually delete the Azure Disk: a. Set the lifecycle on the Azure Disk to delete immediately b.
+ * Try deleting the Azure Disk c. If delete succeeds, finish step d. If delete fails, sleep one
+ * hour; goto (either a or b; maybe a for belts and suspenders)
+ *
+ * <p>As this may take hours to days to complete, this step should never run as part of a
+ * synchronous flight.
+ */
+// TODO: when Stairway implements timed waits, we can use those and not sit on a thread sleeping
+//  for three days.
```

Comment:
I think this comment is cut'n'paste and does not apply to this code. It looks to me like the call to `deleteById()` is synchronous relative to this thread and it will either succeed or fail.

Looks like the error message is cut'n'paste mistake also.